NDEV-1837 Move composability to develop

s-medvedev opened the pull request.

Commit:
NDEV-1837 Move composability to develop branch

Commit:
NDEV-1837 Fix gas tank integration tests

Commit:
Fix tests

afalaleev commented on proxy/common_neon/erc20_wrapper.py:
```diff
@@ -155,7 +155,7 @@ def create_claim_to_ix(self, owner: SolPubKey,
             amount
         ).build_transaction(
-            {'nonce': nonce, 'gasPrice': 0}
+            {'nonce': nonce, 'gasPrice': 0, 'from': signer_acct.address}
```
Ethereum transaction doesn't have 'from'-property.
It is calculated from the signature.
The signature is formed from the private key.
This modification is surplus.

s-medvedev replied:
This change for eth_estimateGas. Without specifying the `from` field, the proxy simulates a transaction with a default `0x000...000` sender and fails due to missing authority. This authority is created from the `contract` and `from` fields.

Commit:
Fix the detection of SolanaCall mode

afalaleev reviewed (commented):
Please squash commits.
I'm ready to approve the PR

s-medvedev committed:
NDEV-2806 Overrides solana accounts (#299)

NDEV-2806 Implement Solana accounts overrides for gas estimation

1. Introduced `SolanaOverrides` as a `Dict[SolPubKey, SolAccountData]`
2. Introduced `EmulateParams` structure with `tag: BlockIdentifier, solana_overrides: SolanaOverrides`
3. Implemented `neon_estimateGas` method which estimates gas using the optional `EmulateParams`
4. Added the optional `EmulateParams` parameter to the `neon_emulate` method

kristinaNikolaevaa committed:
Merge branch 'develop' into NDEV-1837-composability-for-develop

kristinaNikolaevaa committed:
Merge branch 'develop' into NDEV-1837-composability-for-develop
